Purchases/Public/RCPurchases.m

```diff
@@ -910,22 +910,28 @@ - (void)handleOfferingsUpdateError:(NSError *)error completion:(RCReceiveOfferin
     CALL_IF_SET_ON_MAIN_THREAD(completion, nil, error);
 }
 
-- (void)receiptData:(void (^ _Nonnull)(NSData * _Nonnull data))completion
-{
+- (void)receiptData:(RCReceiveReceiptDataBlock)completion {
+    [self receiptDataWithForceRefresh:NO completion:completion];
+}
+
+- (void)receiptDataWithForceRefresh:(BOOL)forceRefresh completion:(RCReceiveReceiptDataBlock)completion  {
     NSData *receiptData = [self.receiptFetcher receiptData];
-    if (receiptData == nil) {
+    if (receiptData == nil || receiptData.length == 0) {
```

Comment:
I can't really think of a scenario where we would want to treat an empty receipt differently than a missing one. It might be a nice simplification to have the fetcher return nil if the data is empty. or even better, some form of an error. 
Again, not for this PR.

Reply:
yeah, I thought the same, right now the fetcher makes the data empty when the receipt is nil. And then there's some error handling in RCPurchases.m (like in the restore function) that checks if the data is empty instead of nil and then sends an error. I thought about moving the error to the fetcher, but it was going to make this PR bigger. I totally agree. I will add it to my list of things to refactor

Reply:
I might tackle this as a part of local receipt validation, since I may have to do some new receipt fetching logic.